nucleus/embedding_index.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from enum import Enum
+
+from nucleus.constants import (
+    EMBEDDING_TYPE_KEY,
+    ID_KEY,
+    INDEX_LEVEL_KEY,
+    INDEX_TYPE_KEY,
+    STATUS_KEY,
+)
+
+
+class IndexType(str, Enum):
+    INTERNAL = "Internal"
+    CUSTOM = "Custom"
+
+
+class IndexLevel(str, Enum):
+    IMAGE = "Image"
+    OBJECT = "Object"
+
+
+class IndexStatus(str, Enum):
+    STARTED = "Started"
+    COMPLETED = "Completed"
+    ERRORED = "Errored"
+
+
+@dataclass
+class EmbeddingIndex:
+    """Represents an Embedding Index belonging to a Dataset.
+
+    Embedding Indexes contain generated embeddings for each item in the dataset,
+    and are used by the Autotag and the Similarity Search functionality.
+    """
+
+    id: str
+    status: IndexStatus
+    index_type: IndexType
+    index_level: IndexLevel
+    embedding_type: str
```

Comment:
Would it make sense to create an enum for this one as well? I was hesitant because it contains valid values like `EfficientNetB7` but also arbitrary ones like `unknown`

Reply:
we can try w/o it for now